src/setupWorker/start/createStart.ts

```diff
@@ -34,10 +34,14 @@ export const createStart = (context: ComposeMocksInternalContext) => {
         return null
       }
 
-      navigator.serviceWorker.addEventListener(
-        'message',
-        handleRequestWith(context, resolvedOptions),
-      )
+      const handleRequest = handleRequestWith(context, resolvedOptions)
+      navigator.serviceWorker.addEventListener('message', handleRequest)
+
+      context.listeners.push({
```

Comment:
Since you've introduced `context.removeAllListeners`, do you think it makes sense to also encapsulate attaching a listener in `context.addListener`?

```ts
context.addListener(navigator.serviceWorker, 'message', handleRequest)
```

- Shorter syntax
- Exact implementation is abstracted (whether we collect those listeners in some array, or operate differently)

I think `context.addListener` should also _attach_ the listener:

```js
{
  addListener(target, event, handler) {
    // Attach the actual listener
    target.addEventListener(event, handler)

    // Push its record, so we can remove it later
    this.listeners.push({})
  }
}
```

Reply:
Yes we can do, I didn't do it before to not refactor everything, but it sounds good